### PR TITLE
Replace LoDTensor with phi::DenseTensor in fluid\operators\*\ except sequence_ops

### PR DESCRIPTION
**Pull request by Liyulingyue**
### PR types

Others 
### PR changes

Others 
### Describe

Replace LoDTensor with phi::DenseTensor in fluid\operators\*\ except sequence_ops
### Related links
- https://github.com/PaddlePaddle/Paddle/issues/48047
- [飞桨 DenseTensor 概念统一](https://github.com/PaddlePaddle/community/blob/master/pfcc/call-for-contributions/tensor_concept_unification.md)
- #46432
- #46663
- #46953

### CODE REVIEW

**Review by YuanRisheng — APPROVED** [review]

**Comment by luotao1** [comment]
@Ligoml 请 review 下 API 部分

File: paddle/fluid/operators/detection/iou_similarity_op.cc [hunk 1]
```diff
@@ -59,17 +59,18 @@ class IOUSimilarityOp : public framework::OperatorWithKernel {
...
-             "Box list X is a 2-D LoDTensor with shape [N, 4] holds N boxes, "
-             "each box is represented as [xmin, ymin, xmax, ymax], "
-             "the shape of X is [N, 4]. [xmin, ymin] is the left top "
-             "coordinate of the box if the input is image feature map, they "
-             "are close to the origin of the coordinate system. "
-             "[xmax, ymax] is the right bottom coordinate of the box. "
-             "This tensor can contain LoD information to represent a batch "
-             "of inputs. One instance of this batch can contain different "
-             "numbers of entities.");
+    AddInput(
```
Comment (luotao1) [inline paddle/fluid/operators/detection/iou_similarity_op.cc:62]:
```
In addition, mistakes found in sample codes: samplecode_temp/paddle.fluid.layers.detection.iou_similarity_example.py
```
@Ligoml 这里需要 review 下

**Review by Ligoml — APPROVED** [review]
LGTM for docs

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**